README.md

```diff
@@ -1,3 +1,221 @@
 # Bash Commons
 
-[WIP]
\ No newline at end of file
+This repo contains a collection of reusable Bash functions for handling common tasks such as logging, assertions,
+string manipulation, and more. It is our attempt to bring a little more sanity, predictability, and coding reuse to our
+Bash scripts. All the code has thorough automated tests and is packaged into functions, so you can safely import it
+into your bash scripts using `source`.
+
+
+
+
+## Examples
+
+Once you have `bash-commons` installed (see the [install instructions](#install)), you use `source` to import the
+modules and start calling the functions within them:
+
+```bash
+source /opt/gruntwork/bash-commons/log.sh
+source /opt/gruntwork/bash-commons/assert.sh
+source /opt/gruntwork/bash-commons/os.sh
```

Comment:
Should we version these folders? Different modules are going to need to refer to specific versions of bash functions and we don't want to cause breakage due to a bash-commons change.

Reply:
No, I definitely don't want to update all sorts of version imports. The `gruntwork-install` call will be versioned. Hopefully, that's enough for now.